### PR TITLE
[FLINK-21329][tests] Increase timeout and delay in local test_local_recovery_and_scheduling.sh

### PR DESCRIPTION
**Pull request by rkhachatryan**
## What is the purpose of the change

1. Fix `test_local_recovery_and_scheduling.sh` by increasing test timeout and add emit interval at sources
1. Ease further debug by logging the time checkpoint mail spent in the mailbox

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

File: flink-end-to-end-tests/run-nightly-tests.sh
Comment (akalash) [inline flink-end-to-end-tests/run-nightly-tests.sh:258]: I actually don't fully get the idea here. As I understand,  after adding the delay the backpressure will be decreased. But won't it decrease the test effectiveness? I mean right now it tests something under high pressure but after this changes, the pressure will be decreased. Perhaps, it is ok because it is not a stress test but it is exactly why I ask.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java
Comment (akalash) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java:78]: Should it be connected somehow with failure timeout(or something similar I don't know the exact name)?

**Review by akalash — COMMENTED** [review]
@rkhachatryan LGTM. I have left a couple of minor comments there. It is up to you to fix them or left as is. Also, please, take a look at my question about delay it is a single thing that I want to understand before approval.

File: flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java
Comment (rkhachatryan) [reply flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java:1289]: Yes, but I wanted to make it explicit.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/checkpointing/CheckpointBarrierHandler.java
Comment (rkhachatryan) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/checkpointing/CheckpointBarrierHandler.java:114]: Same answer: wanted to make it explicit :)

File: flink-end-to-end-tests/run-nightly-tests.sh
Comment (rkhachatryan) [reply flink-end-to-end-tests/run-nightly-tests.sh:258]: Yes, the change decreases back-pressure. But the the test doesn't depend on back-pressure: it creates the state (on init), does some processing, fails, and then checks on recovery whether the state was distributed correctly.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java
Comment (rkhachatryan) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java:78]: Exceeding this threshold is not a failure but rather a potential problem that the user should be warned of.
So I think existing timeouts aren't related to this one.

**Review by akalash — APPROVED** [review]
LGTM

**Comment by rkhachatryan** [comment]
Thanks for the review. 
I squashed the commits and rebased to master, will merge once CI passes.

**Comment by rkhachatryan** [comment]
CI failure unrelated (FLINK-22085, without the fix), merging.

**Pull request merged**
